app/models/support/requests/time_constraint.rb

```diff
@@ -24,6 +24,24 @@ class TimeConstraint
       validates_time :not_before_time, on_or_before: :needed_by_time, unless: proc { |c|
         [c.needed_by_date, c.needed_by_time, c.not_before_date, c.not_before_time].any?(&:blank?) || c.needed_by_date != c.not_before_date
       }
+
+      # def not_before
+      #   date_args = [not_before_day, not_before_month, not_before_year].map(&:to_i)
+      #   begin
+      #     Date.new(*date_args)
+      #   rescue ArgumentError, RangeError
+      #     Struct.new(:day, :month, :year).new(day, month, year)
+      #   end
+      # end
```

Comment:
```suggestion
      def needed_by_date
        return "" unless needed_by_year.present? && needed_by_month.present? && needed_by_day.present?

        date_args = [needed_by_year, needed_by_month, needed_by_day].map(&:to_i)
        date = Date.new(*date_args)

        "#{date.strftime('%d')}-#{date.strftime('%m')}-#{date.strftime('%Y')}"
      end
```
You will then need to update the tests. Similar to what you've done before.